### PR TITLE
refactor(learn): migrate to Nuxt 3

### LINKED ISSUE #3030: Vue 3 update: Migrate learn pages

### PR DESCRIPTION
**Pull request by eddybrando**
## Changes

This PR makes the necessary changes to all "learn" pages to make them work with Nuxt 3.

Closes #3030 

## How to read this PR

Preview links to check:

- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/intro-qc-qh
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/basics-quantum-information
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/games-and-demos
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/introduction-course
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/machine-learning-course
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/multiple-qubits-and-entanglement
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/prerequisites
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/quantum-algorithms-for-applications
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/quantum-computing-labs
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/quantum-hardware-circuits
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/quantum-hardware-pulses
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/quantum-protocols-and-quantum-algorithms
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/course/quantum-states-and-qubits
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/summer-school/introduction-to-quantum-computing-and-quantum-hardware-2020
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/summer-school/quantum-computing-and-quantum-learning-2021
- https://qiskit-org-pr-3037.dcq4xc5i083.us-south.codeengine.appdomain.cloud/learn/summer-school/quantum-simulation-summer-school-2022

### CODE REVIEW

File: components/learn/StartLearningSection.vue [hunk 1]
```diff
@@ -1,109 +1,128 @@
...
+        aria-labelledby="tab-teach"
+        role="tabpanel"
+        hidden
+      >
+        <article
+          v-for="{ title, syllabi } in teachingSections"
+          :key="title"
+          class="cds--row"
+        >
+          <div class="cds--col-xlg-4 cds--col-lg-4">
```
Comment (techtolentino) [inline components/learn/StartLearningSection.vue:74]: Correct me if I'm wrong, but this could be updated to just `cds--col-lg-4`

**Review by techtolentino — CHANGES_REQUESTED** [review]
Really good migration work here @eddybrando 

Just a couple of comments, but nothing major... 

Thank you!

File: components/learn/StartLearningSection.vue [hunk 2]
```diff
@@ -1,109 +1,128 @@
...
+                  title: courseTitle,
+                  subtitle: courseSubtitle,
+                  cta,
+                } in courses"
+                :key="courseTitle"
+                class="cds--col-xlg-8"
+              >
+                <UiAppCard
+                  :cta-label="cta.label"
+                  :to="cta.url"
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply components/learn/StartLearningSection.vue]: It was a bug indeed.

In  **/components/ui/BasicLink.vue* we had `:target="isExternal && '_blank'"`, which compiles to `<a ... target="false">`, causing the link opening in a new tab for an external URLs.

Why these are "external" URLs is another issue, but it's related to the `isExternal` logic, which we can/should change at a later point when we start leveraging the power of the new version of `NuxtLink`. But is a general issue that we already have in the application, so it's out of scope.

Changing the code to `:target="isExternal ? '_blank' : undefined"` makes it work as expected.

Fixed in https://github.com/Qiskit/qiskit.org/pull/3037/commits/02524b485c15b2cdaf1bb27126745104844c7ad7.

[the remark this answers is not in the document]
Comment (techtolentino) [reply components/learn/StartLearningSection.vue]: thanks for explaining this... this is good to know!

**Pull request merged**